src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/StaticClassConstructionContext.cs

```diff
@@ -16,6 +16,8 @@ namespace System.Runtime.CompilerServices
     public struct StaticClassConstructionContext
     {
         // Pointer to the code for the static class constructor method. Set to 0 once the cctor has run.
+        // Volatile to insert memory barriers to order any writes executed as part of static class
+        // constructor with respect to this flag.
         public volatile IntPtr cctorMethodAddress;
```

Comment:
If I'd writing this from scratch, I'd use ordinary field and `Volatile.Read` `Volatile.Write` when necessary while accessing it as that is more self-documenting. This would work too.

Reply:
I agree that it would look better with `Volatile.Read/Write`.